### ISSUE TITLE
Send_Input information is not stored when saving / loading state

### ISSUE THREAD

**Issue by DavidSteinhart** [Baekalfen/PyBoy#170]
When loading a save state, key presses are not stored or reset, so when loading a state and performing input, you will get different results depending on the inputs that were performed before. 

Example:

```
from pyboy.utils import WindowEvent
import pyboy as pyboy
import io


def save_state(boy):
	file_like_object = io.BytesIO()
	file_like_object.seek(0)
	boy.save_state(file_like_object)
	return file_like_object


def load_state(boy, file_like_object):
	file_like_object.seek(0)
	boy.load_state(file_like_object)

# Load Super Mario Land and progress to start of game
boy = pyboy.PyBoy('../SuperMarioLand1.gb')
for i in range(100):
	boy.tick()
boy.send_input(WindowEvent.PRESS_BUTTON_START)
boy.tick()
boy.send_input(WindowEvent.RELEASE_BUTTON_START)
for i in range(20):
	boy.tick()

# Save state of the emulator
state = save_state(boy)

# Send input and progress game
boy.send_input(WindowEvent.PRESS_ARROW_RIGHT)
for i in range(30):
	boy.tick()

# Load state and progress game
load_state(boy, state)
for i in range(30):
	boy.tick()
# Mario keeps running because PRESS_ARROW_RIGHT is still active but it would be expected that he does not move.
```

```
# Use this to reset all input as a quick-fix
for key in range(WindowEvent.RELEASE_ARROW_UP, WindowEvent.RELEASE_BUTTON_START + 1):
	boy.send_input(key)
```

**Comment by Baekalfen** [comment]
Nice find. I guess `interaction.py` might not be covered by the save/load state at all. Although it should of course.

**Comment by R0cker77** [comment]
Hi, i ran the code sample you gave, and mario stopped after the code stopped excuting, so i dont understand what is the issue here, is it the state itself not stopping? appreciate any help..

**Comment by krs013** [comment]
Hi @R0cker77, sorry we never followed up with you on this; I only finally got a good look at this issue now. When I ran the sample code, Mario moved once, reset to his original position, and moved again, which is the wrong behavior. If that wasn't the case when you tried it, I'm not sure why that would be. I think my fix for this should be enough to take care of it, so there's no need to worry about it any more, and thanks for your interest in helping!
